src/SimpleSAML/Utils/HTTP.php

```diff
@@ -229,36 +230,15 @@ private function redirect(string $url, array $parameters = []): void
             Logger::warning('Redirecting to a URL longer than 2048 bytes.');
         }
 
-        if (!headers_sent()) {
-            // set the location header
-            header('Location: ' . $url, true, 303);
+        $response = new RedirectResponse($url, 303);
 
+        if (!headers_sent()) {
             // disable caching of this response
-            header('Pragma: no-cache');
-            header('Cache-Control: no-cache, no-store, must-revalidate');
+            $response->headers->set('Pragma', 'no-cache');
+            $response->headers->set('Cache-Control', 'no-cache, no-store, must-revalidate');
```

Comment:
Better to use `$response->setCache(...)` instead

Reply:
Or even better just drop it... the 303 already documents this is a temporary redirect.

Reply:
You think I can drop the Pragma-header? I read it's HTTP/1.0 specific ..

Reply:
The pragma header is automatically handled by Symfony if you use the setCache() method I suggested. So better leave this to symfony if adding any caching headers at all. Or just skip it. FWIW OC EngineBlock does not add explicit caching headers to redirects.